.github/workflows/nightly.yml

```diff
@@ -0,0 +1,100 @@
+name: Nightly Builds
```

Comment:
This duplicates a lot of logic, wondering if nightly can't be implemented in build.yml?

Reply:
Originally, I had things separated out in different workflows before I realized that github had almost zero support for coordinating things between workflows. As such, were I to redo it today, I'd have a single workflow file to not duplicate the logic. Unfortunately, that can end up with a lot of complex logic around when to skip/run jobs if we're not careful.

But yeah, it makes sense to eventually roll this into build.yml (and likely rename it to ci.yml).